### ISSUE TITLE
Listener remotedeviceready not removed when peer disconnects

### ISSUE THREAD

**Issue by ScottErholm** [zettajs/zetta#375]
Listeners on 'remotedeviceready' are never removed when the initiating peer disconnects, causing memory leak.  MaxListenersExceededWarning issued after default 10 listeners are enabled.

Somewhat related to #294 and could be fixed at the same time as that.

**Comment by AdamMagaluk** [comment]
Do you have a quick example you can provide?

**Comment by ScottErholm** [comment]
Yep, in runtime.js in the 'observe' method, there are two places where the listener is created:
```
self.on(peer.name + '/remotedeviceready', function(device) { ... }
```

But that listener is never removed, even if the peer disconnects.  Then if the same peer reconnects, a new listener is registered. The listeners just keep building up.

I tested moving the listener function out to a function named 'matchDevice' and then subscribing to a disconnect:
```
self.pubsub.subscribe('_peer/disconnect', function(ev, data) {
  if (data.peer.name === query.server) {
    self.removeListener(data.peer.name + '/remotedeviceready', matchDevice);
  }
});
```
